src/net/sourceforge/kolmafia/request/RelayRequest.java

```diff
@@ -3500,6 +3514,156 @@ public boolean pollForCompletion(String type) {
     return true;
   }
 
+  private void jsonError(String message) {
+    this.statusLine = "HTTP/1.1 400 Bad Request";
+    this.contentType = "application/json";
+    this.responseCode = 400;
+    this.responseText = JSON.toJSONString(Map.of("error", message));
+  }
+
+  private String underscoreName(String name) {
+    return name.replaceAll("[A-Z]", "_$0").toLowerCase();
+  }
+
+  private Value transformApiArgument(Object thing) {
+    if (thing instanceof JSONObject obj) {
+      var objectTypeObject = obj.get("objectType");
+      Type dataType;
+      if (objectTypeObject instanceof String objectType
+          && ((dataType = DataTypes.enumeratedTypes.find(objectType.toLowerCase())) != null)) {
+        var identifierStringObject = obj.get("identifierString");
+        var identifierNumberObject = obj.get("identifierNumber");
+        if (identifierStringObject instanceof String identifierString) {
+          return dataType.parseValue(identifierString, true);
+        } else if (identifierNumberObject instanceof Integer identifierNumber) {
+          return dataType.makeValue(identifierNumber, true);
```

Comment:
the way this currently works, a user can pass both identifierString and identifierNumber, and if they conflict it goes for string. I'd like conflicts to go to number (and number to be checked first, for perf). Alternatively, you could error.

Either way, this behaviour should be documented.